### PR TITLE
LPS-124616 override onInstanceReady event callback

### PR DESCRIPTION
**Pull request by marcosapmf**
Hi, 
I needed to make a little change in ClassicEditor component to fix an issue and I'd be very glad if you guys could evaluate that.

Basically, the issue was happening because the following code snippet was setting the wrong value into ckeditor component.
```
editor.on('instanceReady', () => {
    editor.setData(contents);
});
```

My solution consists in overriding this event callback with onIntanceReady function that is passed as a prop. Once onIntanceReady
is not passed, the original event callback will be called.

The related issue is that: https://issues.liferay.com/browse/LPS-124616

Thanks in advance.

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js [hunk 1]
```diff
@@ -54,6 +54,10 @@ const RichText = ({
 						onChange({}, data);
 					}
 				}}
+				onInstanceReady={({editor}) => {
+					editor.setData(currentValue);
+					setCurrentValue(currentValue);
```
Comment (jbalsas) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:59]: Why do you need to `setCurrentValue` here? Is there a chance it has changed?

**Comment by julien** [comment]
@jbalsas sure I can do that, it makes sense.  (I'm thinking about updating to the latest `ckeditor4-react` version too).

**Comment by marcosapmf** [comment]
> Hey @marcosapmf, so you're basically just doing the same but passing it as a prop. Wouldn't just passing the prop to the `Editor` component from `ClassicEditor` just work?
> 
> You're basically passing `currentValue` which is what we read as contents and is set anyways, so not sure why we need the additional level of indirection.
> 
> @julien, could you try to apply your idea of passing both `onInstanceReady` and `onDrop` instead of manually attaching the events to see if that directly fixes this issue?
> 
> Let's try to keep complexity to a minimum here and avoid an explosion of configuration and options since that can make this component hard to reason with

Hi @jbalsas,
I'm passing `currentValue` because despite of `contents` has the right value, somehow the editor data is set with blank on ClassicEditor internal onInstanceReady callback. However, this only occurs on locale changes. Forcing  `currentValue` here was the way I found to around this behavior

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js (hunk 1 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:59]: Yes, there is. When we change the form locale, we need to set the value regard to that locale

**Comment by julien** [comment]
Hi again @marcosapmf,

I submitted #609 which passes the event handlers as props, as well as the `name`.
Please let me know what you think about it.

**Comment by marcosapmf** [comment]
Hey @julien,

I'm going to take a look at that.

Thank you very much!

**Comment by marcosapmf** [comment]
@julien @jbalsas  #609 fixes my issue. Therefore, I go ahead and close this pull.

Thank you guys!

**Comment by jbalsas** [comment]
>@julien @jbalsas #609 fixes my issue. Therefore, I go ahead and close this pull.
>
>Thank you guys!

Awesome!! 👍

**Comment by julien** [comment]
Thanks for confirming @marcosapmf

**Pull request closed without merging**